src/util-file-swf-decompression.c

```diff
@@ -34,7 +34,7 @@
 
 #include <zlib.h>
 
-#include <htp/lzma/LzmaDec.h>
+#include <htp/htp.h>
```

Comment:
@victorjulien was there the idea of deprecating and removing this whole code ?

Reply:
The LZMA decompression functionality was moved to [lzma-rs](https://crates.io/crates/lzma-rs), so the set of events is not 1-1 that come from that decompression library. At the time, we had discussed how to best do this, and sent some patches upstream to `lzma-rs` to add some checks and features that would make it acceptable to use in suricata. Once those were merged we moved over to lzma-rs and removed the libhtp lzma code since it wasn't needed anymore.

I think a better solution is to just have suricata depend on lzma-rs directly, rather than having libhtp export some LZMA functions. The [code](https://github.com/cccs-rtmorti/libhtp-rs/blob/master/src/c_api/lzma.rs) to do the decompression is not very heavy, so it doesn't really make that much difference where it lives, but it might be good to separate http parsing and lzma decompression. Up to you.

Reply:
> I think a better solution is to just have suricata depend on lzma-rs directly

I agree. That could be a commit before switching to libhtp-rs